### DISCUSSION TITLE
Does rust-analyzer.procMacro.ignored work with function-like proc macros?

### DISCUSSION THREAD

**Discussion by basro** [rust-lang/rust-analyzer#16316] [Q&A]
I've implemented a crate that reexports a function-like proc macro from an internal crate.

I don't want rust analyzer to expand the proc macro so I tried setting 
```json
"rust-analyzer.procMacro.ignored": {
   "my_crate": ["my_function_like_proc_macro"]
},
```
and even:
```json
"rust-analyzer.procMacro.ignored": {
   "my_internal_crate": ["my_function_like_proc_macro"]
},
```

But to no avail, rust analyzer continues to expand that macro.

All the examples I've found of procMacro.ignored are for attrib like proc macros, so I'm wondering, Is the feature supposed to work with function like proc macros?

**Comment by Veykril** [comment]
Hmm, it should be working (but also not), that is it should give you different errors at least. https://github.com/rust-lang/rust-analyzer/pull/15923 should properly fix that
